src/coreclr/jit/lsraloongarch64.cpp

```diff
@@ -0,0 +1,1727 @@
+    assert(argNode->gtOper == GT_PUTARG_STK);
+
+    GenTree* putArgChild = argNode->gtGetOp1();
+
+    int srcCount = 0;
+
+    // Do we have a TYP_STRUCT argument (or a GT_FIELD_LIST), if so it must be a multireg pass-by-value struct
+    if (putArgChild->TypeIs(TYP_STRUCT) || putArgChild->OperIs(GT_FIELD_LIST))
+    {
+        // We will use store instructions that each write a register sized value
+
+        if (putArgChild->OperIs(GT_FIELD_LIST))
+        {
+            assert(putArgChild->isContained());
+            // We consume all of the items in the GT_FIELD_LIST
+            for (GenTreeFieldList::Use& use : putArgChild->AsFieldList()->Uses())
+            {
+                BuildUse(use.GetNode());
+                srcCount++;
+            }
+        }
+        else
+        {
+            // We can use a ldp/stp sequence so we need two internal registers for LOONGARCH64; one for ARM.
+            buildInternalIntRegisterDefForNode(argNode);
+
+            if (putArgChild->OperGet() == GT_OBJ)
+            {
+                assert(putArgChild->isContained());
+                GenTree* objChild = putArgChild->gtGetOp1();
+                if (objChild->OperGet() == GT_LCL_VAR_ADDR)
+                {
+                    // We will generate all of the code for the GT_PUTARG_STK, the GT_OBJ and the GT_LCL_VAR_ADDR
+                    // as one contained operation, and there are no source registers.
+                    //
+                    assert(objChild->isContained());
+                }
+                else
+                {
+                    // We will generate all of the code for the GT_PUTARG_STK and its child node
+                    // as one contained operation
+                    //
+                    srcCount = BuildOperandUses(objChild);
+                }
+            }
+            else
+            {
+                // No source registers.
+                putArgChild->OperIs(GT_LCL_VAR);
+            }
+        }
+    }
+    else
+    {
+        assert(!putArgChild->isContained());
+        srcCount = BuildOperandUses(putArgChild);
+    }
+    buildInternalRegisterUses();
+    return srcCount;
+}
+
+#if FEATURE_ARG_SPLIT
+//------------------------------------------------------------------------
+// BuildPutArgSplit: Set the NodeInfo for a GT_PUTARG_SPLIT node
+//
+// Arguments:
+//    argNode - a GT_PUTARG_SPLIT node
+//
+// Return Value:
+//    The number of sources consumed by this node.
+//
+// Notes:
+//    Set the child node(s) to be contained
+//
+int LinearScan::BuildPutArgSplit(GenTreePutArgSplit* argNode)
+{
+    int srcCount = 0;
+    assert(argNode->gtOper == GT_PUTARG_SPLIT);
+
+    GenTree* putArgChild = argNode->gtGetOp1();
+
+    // Registers for split argument corresponds to source
+    int dstCount = argNode->gtNumRegs;
+
+    regNumber argReg                  = argNode->GetRegNum();
+    regMaskTP argMask                 = RBM_NONE;
+    regMaskTP argMaskArr[MAX_REG_ARG] = {RBM_NONE};
```

Comment:
I am not sure if there is a need for `argMaskArr`. Code can be simplified.

Reply:
Yes, you are right.
This is old code for LoongArch64.

Thanks

Reply:
Besides I have a question for your help, the argsplit is that a struct which can be passed by registers is passed by register and stack ?

Reply:
If you refer the `#define FEATURE_ARG_SPLIT`, it says it is for "// Case: splitting 16 byte struct between x7 and stack"